tests/python/pants_test/backend/python/tasks/test_select_interpreter.py

```diff
@@ -28,33 +30,40 @@ def setUp(self):
 
         # We're tied tightly to pex implementation details here faking out a python binary that outputs
         # only one value no matter what arguments, environment or input stream it has attached. That
-        # value is the interpreter identity which is - minimally, one line containing:
-        # <impl> <abi> <impl_version> <major> <minor> <patch>
-
-        def fake_interpreter(id_str):
+        # value is the interpreter identity which is a JSON dict with exactly the following keys:
+        # binary, python_tag, abi_tag, platform_tag, version, supported_tags, env_markers.
+        def fake_interpreter(python_tag: str, abi_tag: str, version: Tuple[int, int, int]):
             interpreter_dir = safe_mkdtemp()
             binary = os.path.join(interpreter_dir, "python")
+            values = dict(
+                binary=binary,
+                python_tag=python_tag,
+                abi_tag=abi_tag,
+                platform_tag="",
+                version=version,
+                supported_tags=[],
+                env_markers={},
+            )
```

Comment:
Is there a particular reason to do this instead of a dict literal? Generally, literals have better performance: https://stackoverflow.com/questions/30216000/why-is-faster-than-list

Reply:
Yes. I always use dict literals for their restriction on key names when the key names are fed to other function calls as kwargs. This mirrors the PythonIdentity scheme in Pex: https://github.com/pantsbuild/pex/blob/2befa56b9e6b803e06277d9ca8dbf7472a22b4dc/pex/interpreter.py#L85-L117
I additionally tend to use them to construct environment variable dicts as well since environment variables match the python identifier restrictions well enough: https://pubs.opengroup.org/onlinepubs/000095399/basedefs/xbd_chap08.html